src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -10,6 +10,63 @@
 
 #include <ShellScalingApi.h>
 
+namespace {
+  enum class TTmpFileType {
+    EActiveZoneSetFile = 0,
+    EAppliedZoneSetFile,
+    ECustomZoneSetsFile
+  };
+
+  std::wstring GenerateTmpFileName(TTmpFileType type)
+  {
+    static std::wstring activeZoneSetTmpFileName;
```

Comment:
We should probably use `%TEMP%\fancyzones-active-set.json` etc. here instead of all this temp filename logic.

Reply:
There's also [`tmpfile`](https://en.cppreference.com/w/cpp/io/c/tmpfile) function which basically does all this logic.

Reply:
tmpfile returns std::FILE*, but we need to use same file from c# and cpp, that's why I'm passing the names. Is there a way to do that with std::FILE* and tmpfile?

Reply:
@stefansjfw sure, we could use [`std::tmpnam`](https://en.cppreference.com/w/cpp/io/c/tmpnam).

Reply:
Nice. I agree with @bzoz to keep it simple and name them `fancyzones-active-set.json` , etc.. Only problem would be if there is already file with that name.. But what are the chances...

Reply:
@stefansjfw why randomize it in the first place? 🙂 `fancyzones-active-set.json` could be already there if there's a bug/we've been terminated/other unexpected things.

Reply:
It was temp solution at the beginning and here it is now :) std::tmpnam is the safest solution indeed

Reply:
tmpnam it is. Done